corehq/sql_db/util.py

```diff
@@ -150,3 +153,33 @@ def _get_all_nested_subclasses(cls):
         if subclass not in seen:
             seen.add(subclass)
             yield subclass
+
+
+STANDBY_DELAY_FRESHNESS = 30
+
+
+@quickcache(timeout=STANDBY_DELAY_FRESHNESS)
+def get_replication_delay_for_standby(db_alias):
+    """
+    Finds the replication delay for given database by running a SQL query on standby database.
+        See https://www.postgresql.org/message-id/CADKbJJWz9M0swPT3oqe8f9+tfD4-F54uE6Xtkh4nERpVsQnjnw@mail.gmail.com
+
+    If the given database is not a standby database, an assertion error is raised
+    If standby process (wal_receiver) is not running on standby a `VERY_LARGE_DELAY` is returned
+    """
+    assert db_alias in STANDBY_DATABASE_ALIASES
+    # used to indicate that the wal_receiver process on standby is not running
+    VERY_LARGE_DELAY = 100000
+    sql = """
+    SELECT
+    CASE
+        WHEN NOT EXISTS (SELECT 1 FROM pg_stat_wal_receiver) THEN {delay}
```

Comment:
If wal_receiver is running then `pg_stat_wal_receiver` returns a single row with PID and other columns, otherwise it returns no rows. Couple of notes here
- In the result of `pg_stat_wal_receiver` all columns except PID column are hidden for non-root users, so since we run this as `commcarehq` we only get PID column, other column values are empty.
- I am not 100% sure whether the `state` column from the result of `pg_stat_wal_receiver` should make a difference here (I couldn't find what what possible values it could have except the default `streaming`).

Reply:
The states are listed here: https://www.postgresql.org/docs/10/static/monitoring-stats.html#PG-STAT-REPLICATION-VIEW

I think as long as it's running and the replication time is low we should be OK.